packages/esm-patient-forms-app/src/hooks/use-forms.ts

```diff
@@ -10,17 +10,28 @@ import {
 } from '@openmrs/esm-framework';
 import type { ConfigObject } from '../config-schema';
 import type { ListResponse, Form, EncounterWithFormRef, CompletedFormInfo } from '../types';
-import { customEncounterRepresentation, formEncounterUrl, formEncounterUrlPoc } from '../constants';
+import {
+  customEncounterRepresentation,
+  customFormRepresentation,
+  formEncounterUrl,
+  formEncounterUrlPoc,
+} from '../constants';
 import { isValidOfflineFormEncounter } from '../offline-forms/offline-form-helpers';
 
-export function useFormEncounters(cachedOfflineFormsOnly = false, patientUuid: string = '') {
+export function useFormEncounters(cachedOfflineFormsOnly = false, patientUuid: string = '', visitUuid: string = '') {
   const { customFormsUrl, showHtmlFormEntryForms } = useConfig<ConfigObject>();
   const hasCustomFormsUrl = Boolean(customFormsUrl);
-  const url = hasCustomFormsUrl
-    ? customFormsUrl.concat(`?patientUuid=${patientUuid}`)
+  let url = hasCustomFormsUrl
+    ? customFormsUrl.indexOf('?') === -1
+      ? `${customFormsUrl}?patientUuid=${patientUuid}`
+      : customFormsUrl
     : showHtmlFormEntryForms
-    ? formEncounterUrl
-    : formEncounterUrlPoc;
+      ? formEncounterUrl
+      : formEncounterUrlPoc;
+  url = url
+    .replace('{patientUuid}', patientUuid)
+    .replace('{visitUuid}', visitUuid)
+    .replace('{representation}', customFormRepresentation);
```

Comment:
We actually have a function to do this, called `interpolateUrl()` in the framework, which works with `${patientUuid}`-style syntax. I think it would be nicer to consistently use that here.

Reply:
Oh nice, thanks @ibacher - I looked for something like that but didn't find it.  I've updated the code to use this.  Let me know how that looks.